### ISSUE TITLE
Jupyter Lab vs Vega-Lite update cycles

### ISSUE THREAD

**Issue by davidanthoff** [jupyterlab/jupyterlab#4695]
I'm running into a problem that the vega-lite version that ships with Jupyter Lab currently has a bug that seems to be fixed in newer vega-lite versions. 

Is there generally a plan to somehow be able to ship updated vega-lite versions on a more frequent basis than the release cycles that Jupyter Lab has?

**Comment by jasongrout** [comment]
> Is there generally a plan to somehow be able to ship updated vega-lite versions on a more frequent basis than the release cycles that Jupyter Lab has?

No, not currently. Since vega support is a core plugin, it's tied to the core release cycle. We could make it not a core plugin, but that has other downsides.

That said, we've updated to the most recent vega-lite just today in master.

**Comment by jasongrout** [comment]
also, the way our build framework works, installing *any* extension should pull in the most recent compatible vega-lite version. Can you try installing any custom extension and rebuilding JupyterLab?

**Comment by jasongrout** [comment]
In fact, it's possible that `jupyter lab build` may update the vega-lite (and many other) dependency by itself, without having to install another plugin.

**Comment by vidartf** [comment]
Should there be a menu entry / command palette entry to do a fresh build?

**Comment by davidanthoff** [comment]
> We could make it not a core plugin, but that has other downsides.

I completely agree that it being a core plugin is fantastic!

> In fact, it's possible that jupyter lab build may update the vega-lite (and many other) dependency by itself, without having to install another plugin.

Yes, that actually works!

**Comment by bollwyvl** [comment]
One note... I would like to see the Vega dependencies lazy loaded, which webpack can do handily. It's increasing the size of the main bundle by 1mb (60%), which goes right to the bottom line of every single user's initial experience whether or not they Vega.

 I can probably help on this, having grappled with this on drawio and graphviz where the effect is even more pronounced.

**Comment by aldanor** [comment]
Also eagerly waiting for a next jupyterlab release so you can render Vega SVGs which currently doesn't seem doable due to metadata being ignored...

**Comment by jasongrout** [comment]
We're working hard on it!

**Comment by jasongrout** [comment]
Accidentally closed this.
